packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -124,19 +127,88 @@ export class Rule extends Resource implements IRule {
    *
    * No-op if target is undefined.
    */
-  public addTarget(target?: IRuleTarget) {
+  public addTarget(target?: IRuleTarget): void {
     if (!target) { return; }
 
     // Simply increment id for each `addTarget` call. This is guaranteed to be unique.
-    const id = `Target${this.targets.length}`;
+    const autoGeneratedId = `Target${this.targets.length}`;
 
-    const targetProps = target.bind(this, id);
+    const targetProps = target.bind(this, autoGeneratedId);
     const inputProps = targetProps.input && targetProps.input.bind(this);
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
+    const id = targetProps.id || autoGeneratedId;
+
+    if (targetProps.targetResource) {
+      const targetStack = Stack.of(targetProps.targetResource);
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+      if (targetStack.account !== sourceAccount) {
+        // cross-account event - strap in, this works differently than regular events!
+        // based on:
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEvents-CrossAccountEventDelivery.html
+
+        // the _actual_ target is just the event bus of the target's account
+        // make sure we only add it once
+        const exists = this.accountEventBusTargets[targetStack.account];
+        if (!exists) {
+          this.accountEventBusTargets[targetStack.account] = true;
+          this.targets.push({
+            id,
+            arn: targetStack.formatArn({
+              service: 'events',
+              resource: 'event-bus',
+              resourceName: 'default',
+              region: targetStack.region,
+              account: targetStack.account,
+            }),
+          });
+        }
+
+        // Grant the source account permissions to publish events to the event bus of the target account.
+        // Do it in a separate stack instead of the target stack (which seems like the obvious place to put it),
+        // because it needs to be deployed before the rule containing the above event-bus target in the source stack
+        // (CloudWatch verifies whether you have permissions to the targets on rule creation),
+        // but it's common for the target stack to depend on the source stack
+        // (that's the case with CodePipeline, for example)
+        const app = this.node.root;
+        if (!app || !App.isApp(app)) {
+          throw new Error(`Event stack which uses cross-account targets must be part of a CDK app`);
+        }
+        const stackId = `EventBusPolicy-${sourceAccount}-${targetStack.account}`;
```

Comment:
Oops, just realized that the 2 accounts are not enough, we also need the region in there. Will fix.

Reply:
Fixed.